src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs

```diff
@@ -70,18 +70,26 @@ public static string GetRequiredValue(this IConfiguration configuration, string
         builder.AddGitHubClientFactory();
         builder.AddWorkitemQueues(azureCredential, waitForInitialization: initializeService);
 
-        builder.Services.AddSingleton<IMaestroApi>(s =>
+        builder.Services.AddScoped<IMaestroApi>(s =>
         {
             var config = s.GetRequiredService<IConfiguration>();
             var uri = config.GetValue<string>("Maestro:Uri")
                 ?? throw new Exception("Missing configuration key Maestro.Uri");
 
-            // TODO https://dev.azure.com/dnceng/internal/_workitems/edit/6451: Implement Maestro - PCS communication
-            var token = config.GetValue<string>("Maestro:Token");
+            var noAuth = config.GetValue<bool>("Maestro:NoAuth");
+            if (noAuth)
+            {
+                return MaestroApiFactory.GetAnonymous(uri);
+            }
+
+            var managedIdentityId = config.GetValue<string>("ManagedIdentityClientId");
```

Comment:
```suggestion
            var managedIdentityId = config[ManagedIdentityId];
```
we have a const with the key value